dotnet/src/Connectors/Connectors.AI.OpenAI/ImageGeneration/AzureOpenAIImageGeneration.cs

```diff
@@ -0,0 +1,149 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.SemanticKernel.AI;
+using Microsoft.SemanticKernel.AI.ImageGeneration;
+using Microsoft.SemanticKernel.Connectors.AI.OpenAI.CustomClient;
+using Microsoft.SemanticKernel.Diagnostics;
+using Microsoft.SemanticKernel.Text;
+
+namespace Microsoft.SemanticKernel.Connectors.AI.OpenAI.ImageGeneration;
+
+public class AzureOpenAIImageGeneration : OpenAIClientBase, IImageGeneration
+{
+    /// <summary>
+    /// Constants of success status
+    /// </summary>
+    private const string SucceededStatus = "SUCCEEDED";
+
+    /// <summary>
+    /// Maximum retry times for the Azure Dall-E API
+    /// </summary>
+    private const int MaxRetryCount = 5;
```

Comment:
We can add this as optional parameter to constructor and use default value from this constant. We can add the same parameter to extension method in `OpenAIKernelBuilderExtensions`. In this case `max retry count` will be customizable by user.